pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -59,13 +59,20 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(env *model.Environme
 	errs := &multierror.Error{}
 	listeners := make([]*xdsapi.Listener, 0, len(mergedGateway.Servers))
 	for portNumber, servers := range mergedGateway.Servers {
+		var si *model.ServiceInstance
+		for _, w := range workloadInstances {
+			if w.Endpoint.ServicePort.Port == int(portNumber) {
+				si = w
```

Comment:
Missing `break` after this line?

Reply:
Yep, although I think it's innocuous as any matching ServiceInstance will have the same NetworkEndpoint.Port, but certainly less efficient.

Good catch!